policy-service/src/component-tests/java/uk/gov/gchq/palisade/component/policy/service/CaffeinePolicyCachingTest.java

```diff
@@ -15,42 +15,72 @@
 import org.springframework.boot.test.context.SpringBootTest.WebEnvironment;
 import org.springframework.cache.CacheManager;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.ConfigurableApplicationContext;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.Import;
+import org.springframework.context.annotation.Primary;
+import org.springframework.core.env.Environment;
+import org.springframework.core.io.ResourceLoader;
 import org.springframework.test.context.ActiveProfiles;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.support.TestPropertySourceUtils;
+import org.testcontainers.containers.KafkaContainer;
 
-import uk.gov.gchq.palisade.contract.policy.PolicyTestData;
+import uk.gov.gchq.palisade.contract.policy.PolicyTestCommon;
 import uk.gov.gchq.palisade.policy.PassThroughRule;
 import uk.gov.gchq.palisade.resource.Resource;
 import uk.gov.gchq.palisade.resource.StubResource;
 import uk.gov.gchq.palisade.resource.impl.FileResource;
 import uk.gov.gchq.palisade.resource.impl.SystemResource;
 import uk.gov.gchq.palisade.service.SimpleConnectionDetail;
 import uk.gov.gchq.palisade.service.policy.PolicyApplication;
+import uk.gov.gchq.palisade.service.policy.config.ApplicationConfiguration;
 import uk.gov.gchq.palisade.service.policy.service.PolicyServiceCachingProxy;
+import uk.gov.gchq.palisade.service.policy.stream.PropertiesConfigurer;
 import uk.gov.gchq.palisade.service.request.Policy;
 
+import java.util.AbstractMap;
 import java.util.Arrays;
 import java.util.List;
+import java.util.Map;
 import java.util.Objects;
 import java.util.Optional;
+import java.util.concurrent.ExecutionException;
 import java.util.concurrent.TimeUnit;
 import java.util.function.Function;
+import java.util.stream.Stream;
 
+import static java.util.stream.Collectors.toMap;
+import static org.apache.kafka.clients.admin.AdminClientConfig.BOOTSTRAP_SERVERS_CONFIG;
 import static org.assertj.core.api.Assertions.assertThat;
 
-@ActiveProfiles("caffeine")
+@ActiveProfiles({"caffeine", "akkatest"})
+@Import({CaffeinePolicyCachingTest.KafkaInitializer.Config.class})
+@ContextConfiguration(initializers = {CaffeinePolicyCachingTest.KafkaInitializer.class})
 @SpringBootTest(
-        classes = PolicyApplication.class,
+        classes = {PolicyApplication.class, ApplicationConfiguration.class},
         webEnvironment = WebEnvironment.NONE,
-        properties = "spring.cache.caffeine.spec=expireAfterWrite=1s, maximumSize=100"
+        properties = {"spring.cache.caffeine.spec=expireAfterWrite=1s, maximumSize=100"}
 )
-class CaffeinePolicyCachingTest extends PolicyTestData {
+class CaffeinePolicyCachingTest extends PolicyTestCommon {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(CaffeinePolicyCachingTest.class);
```

Comment:
Looks like a legacy item, but we have assertions in the setup method.  Think these should be in a test in their own right.  It is unusual, so if they need to be here is should be explained.

Reply:
was used as an extra step of validation but is not neccessary so removed